python/runtime/cudaq/algorithms/py_run.cpp

```diff
@@ -0,0 +1,168 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include "cudaq/algorithms/run.h"
+#include "utils/OpaqueArguments.h"
+#include "mlir/Bindings/Python/PybindAdaptors.h"
+#include <pybind11/complex.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <iostream>
+
+namespace cudaq {
+
+void pyAltLaunchKernel(const std::string &, MlirModule, OpaqueArguments &,
+                       const std::vector<std::string> &);
+
+namespace {
+
+inline unsigned int byteSize(mlir::Type ty) {
+  if (isa<ComplexType>(ty)) {
+    auto eleTy = cast<ComplexType>(ty).getElementType();
+    return 2 * cudaq::opt::convertBitsToBytes(eleTy.getIntOrFloatBitWidth());
+  }
+  return cudaq::opt::convertBitsToBytes(ty.getIntOrFloatBitWidth());
```

Comment:
There are functions in the core that compute the size of types like structs, etc. This will just return `0`, which will be incorrect in the algorithm below, I'm thinking.

Reply:
I plan to add more to it as we include more return types. Added a check for now.